### ISSUE TITLE
New feature: read persona from files

### ISSUE THREAD

**Issue by chrisrude** [chrisrude/oobabot#25]
In main I've added an option to read persona information from a file.

Hopefully, this should support a few of the common formats,
though it will only read in the name and description, not
other fields.

This is exposed as a new setting in `config.yml`.
```yaml
  # Path to a file containing a persona.
  # This can be just a single string, a json file in
  # the common "tavern" formats, or a yaml file in the
  # Oobabooga format.
  # With a single string, the persona will be set to that string.
  # Otherwise, the ai_name and persona will be overwritten with
  # the values in the file.  Also, the wakewords will be extended to
  # include the character's own name.
  #   default:
  persona_file: ./tests/test_data/test-char-1.yaml
```

If you're interested, please try this out and let me know if it works for you.  Also let me know if there's a burning desire to support character cards (images).  It would be possible to add but would add dependencies on external libraries, so I would rather not pull those in unless there's demand.

More details in commit.
https://github.com/chrisrude/oobabot/commit/6f95d7c9259831b9cc072058e9c7cd04d0c2d2f6

**Comment by chrisrude** [comment]
@Kozonak FYI.  Let me know if this works for you, or if you're using a different format.

I'm a little unsure about which fields exactly need to be read, so it's possible I need to try and incorporate more data from the card.  The bot will show you what it's using when it starts up, so let me know if you think that will work.

**Comment by Kozonak** [comment]
This sounds great, I will update to latest tomorrow and take it for a spin. I was happy with just txt but other formats might help keep things tidy.

**Comment by sidonsoft** [comment]
pip install ruamel.yaml

now required

**Comment by chrisrude** [comment]
Yup!  In the dev environment after an update you can just run `poetry install`, and it will pull in all the needed dependencies.  That way you don't have to figure out what it is missing, and be assured that you're running against the same library versions that the python egg will install.

**Comment by rebek43** [comment]
an odd behavior i've noticed with this is that the bot does only register the persona of the character file while ignoring other possibly relevant sections (e.g. the example dialogue)
though that may be intended

**Comment by johnmoney83748932** [comment]
If you load the same persona png/json in ooba ui, it still only loads the persona itself (as far as i can tell). I think certain params like example history are only parsed in certain situations or maybe only in tavern.

**Comment by chrisrude** [comment]
This is intended, and probably not great.

The reason is that I haven't figured out yet how to prompt the AI with the sample dialog.  I know that other apps do use it, but I haven't dove into that yet, as I've been prioritizing the voice feature.

It's totally possible to add that in as well, if you think it would help.  It would make it easier if I knew what sort of prompts are effective for sending to oobabooga -- i.e. how is the example dialogue included in the prompt?

**Comment by johnmoney83748932** [comment]
> The reason is that I haven't figured out yet how to prompt the AI with the sample dialog.

I've been experimenting with that, and I've added a line to the "persona" part of the prompt that gets sent.
It's something along the lines of 
```
Rose is a cat, blah blah blah blah.
Below is a past conversation between {{char}} and a user:
USER: example message
{{char}}: example response
```
